### PR TITLE
update "working on a crate"

### LINKED ISSUE #14688: Cargo: [replace] statement ignored
Any idea why I am getting: 
```
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#webrender_traits:0.11.0
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#webrender:0.11.0
```

I've added the following statement in the workplace `Cargo.toml` here https://github.com/servo/servo/blob/master/Cargo.toml

```
[replace]
"webrender:0.11.0" = { path = '/Users/gregory/Projects/webrender/webrender' }
"webrender_traits:0.11.0" = { path = '/Users/gregory/Projects/webrender/webrender_traits/' }```

### PR DESCRIPTION
**Pull request by gterzian**
Update of the "how to replace a dependency" part of "Hacking Servo". 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14688 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it only affects documentation.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14701)

### CODE REVIEW

File: docs/HACKING_QUICKSTART.md [hunk 1]
```diff
@@ -108,20 +108,28 @@ This is how my projects are laid out:
 
 ```
 ~/my-projects/servo/
-~/my-projects/cocoa-rs/
-~/my-projects/glutin/
+~/my-projects/webrender/webrender
```
Comment (SimonSapin) [inline docs/HACKING_QUICKSTART.md]: I *think* that you mean for `~/my-projects/webrender` to be the root of the git repository, and the `webrender` directory inside of it contains `Cargo.toml` for that crate. But that’s not very clear. To make it consistent with the line above (which is `~/my-projects/servo`, not `~/my-projects/servo/ports/servo`), maybe change this line to just `~/my-projects/webrender`?

File: docs/HACKING_QUICKSTART.md [hunk 2]
```diff
@@ -108,20 +108,28 @@ This is how my projects are laid out:
...
-These are all git repositories.
+To make it so that servo uses `~/my-projects/webrender/webrender`, first ascertain which version of the crate Servo is using and whether it is a git dependency or one from crates.io.  
 
-To make it so that servo uses `~/my-projects/cocoa-rs/` and `~/my-projects/glutin/` , create a `~/my-projects/servo/.cargo/config` file:
+Both information can be found using, in this exampple, `/mach cargo pkgid webrender`.
 
-``` shell
-$ cat ~/my-projects/servo/.cargo/config
-paths = ['../glutin', '../cocoa-rs']
+If the output is in the format `https://github.com/servo/webrender#0.11.0`, you are dealing with a git dependency and you will have to edit the `~/my-projects/servo/Cargo.toml` file and add at the bottom:
```
Comment (SimonSapin) [inline docs/HACKING_QUICKSTART.md]: Is `webrender:` missing between `#` and `0.11.0` here?

Comment (gterzian) [inline docs/HACKING_QUICKSTART.md]: nope, that's  what `./mach cargo pkgid webrender` prints, although indeed in the `replace` statement one needs to ad a `:`

File: docs/HACKING_QUICKSTART.md (hunk 1 above)
Comment (gterzian) [inline docs/HACKING_QUICKSTART.md]: yep you're right

File: docs/HACKING_QUICKSTART.md (hunk 2 above)
Comment (gterzian) [inline docs/HACKING_QUICKSTART.md]: actually for other packages it prints, for example `https://github.com/servo/mozjs#mozjs_sys:0.0.0` so indeed with a `:`...

**Comment by gterzian** [comment]
I'm wondering if webrender is actually the right example, since the crate is actually a sub-folder of the repo, which contains two crates.. maybe use a more simple example?

**Comment by gterzian** [comment]
Ok I switched to mozjs in the example, because otherwise with webrender it's not clear why the `path` in `[replace]` is `../webrender/webrender` and not just `../webrender`

**Comment by SimonSapin** [comment]
Yes, good example with a crate name different from the package name. Looks good, thanks!

@bors-servo r+

**Comment by gterzian** [comment]
@SimonSapin thanks for the review, it has been squashed...

**Pull request merged**